morebits.js

```diff
@@ -1308,29 +1308,238 @@ Morebits.unbinder.getCallback = function UnbinderGetCallback(self) {
+	},
+	getUTCDayNameAbbrev: function() {
+		return Morebits.date.localeData.daysShort[this.getUTCDay()];
+	},
+	getDayName: function() {
+		return Morebits.date.localeData.days[this.getDay()];
+	},
+	getDayNameAbbrev: function() {
+		return Morebits.date.localeData.daysShort[this.getDay()];
+	},
+
+	/**
+	 * Add a given number of minutes, hours, days, months or years to the date.
+	 * This is done in-place. The modified date object is also returned, allowing chaining.
+	 * @param {number} number
+	 * @param {string} unit
+	 * @throws {Error} if invalid or unsupported unit is given
+	 * @returns {Morebits.date}
+	 */
+	add: function(number, unit) {
+		// mapping time units with getter/setter function names
+		var unitMap = {
+			minutes: 'Minutes',
+			hours: 'Hours',
+			days: 'Date',
+			months: 'Month',
+			years: 'FullYear'
+		};
+		var unitNorm = unitMap[unit] || unitMap[unit + 's']; // so that plural forms also work
+		if (unitNorm) {
+			this['set' + unitNorm](this['get' + unitNorm]() + number);
+			return this;
+		}
+		throw new Error('Invalid unit "' + unit + '": Only ' + Object.keys(unitMap).join(', ') + ' are allowed.');
+	},
+
+	/**
+	 * Subtracts a given number of minutes, hours, days, months or years to the date.
+	 * This is done in-place. The modified date object is also returned, allowing chaining.
+	 * @param {number} number
+	 * @param {string} unit
+	 * @throws {Error} if invalid or unsupported unit is given
+	 * @returns {Morebits.date}
+	 */
+	subtract: function(number, unit) {
+		return this.add(-number, unit);
+	},
+
+	/**
+	 * Formats the date into a string per the given format string.
+	 * Replacement syntax is a subset of that in moment.js.
+	 * @param {string} formatstr
+	 * @param {(string|number)} [zone=system] - 'system' (for browser-default time zone), 'utc' (for UTC),
+	 * or 'user' (for time zone according to user's site preferences, needs mediawiki.user dependency), or
+	 * specify a time zone as number of minutes past UTC
+	 * @returns {string}
+	 */
+	format: function(formatstr, zone) {
+		var udate = this;
+		// create a new date object that will contain the date to display as system time
+		if (zone === 'utc') {
+			udate = new Morebits.date(this.getTime()).add(this.getTimezoneOffset(), 'minutes');
+		} else if (zone === 'user') {
+			var offset = parseInt(mw.user.options.get('timecorrection').split('|')[1]);
+			udate = new Morebits.date(this.getTime()).add(offset, 'minutes');
+		} else if (typeof zone === 'number') {
+			// convert to utc, then add the utc offset given
+			udate = new Morebits.date(this.getTime()).add(this.getTimezoneOffset() + zone, 'minutes');
+		}
+
+		var pad = function(num) {
+			return num < 10 ? '0' + num : num;
+		};
+		var h24 = udate.getHours(), m = udate.getMinutes(), s = udate.getSeconds();
+		var D = udate.getDate(), M = udate.getMonth() + 1, Y = udate.getFullYear();
+		var h12 = h24 > 12 || h24 === 0 ? Math.abs(h24 - 12) : h24;
```

Comment:
I like the use of `abs` here, but couldn't we just do:
```suggestion
		var h12 = h24 % 12 || 12;
```

Reply:
Yep. `for (var i = 0; i <=23; i++) console.log(`${i}: ${i % 12 || 12}`);` confirms that's correct. Thanks!